src/api/VoteMonitor.Api.Observer/Handlers/RemoveDeviceIdHandler.cs

```diff
@@ -0,0 +1,53 @@
+﻿using MediatR;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using VoteMonitor.Api.Observer.Commands;
+using VoteMonitor.Entities;
+
+namespace VoteMonitor.Api.Observer.Handlers
+{
+    public class RemoveDeviceIdHandler : IRequestHandler<RemoveDeviceIdCommand, bool>
+    {
+        private readonly VoteMonitorContext _voteMonitorContext;
+        private readonly ILogger _logger;
+
+        public RemoveDeviceIdHandler(VoteMonitorContext context, ILogger<ResetDeviceHandler> logger)
+        {
+            _voteMonitorContext = context;
+            _logger = logger;
+        }
+
+        public Task<bool> Handle(RemoveDeviceIdCommand request, CancellationToken cancellationToken)
+        {
+            try
+            {
+                var observerQuery = _voteMonitorContext.Observers
```

Comment:
lines 27 to 35 could be simplified to a single one if FirstAsync is used. If FirstAsync doesn't find any observer, then it will throw an exception so no need to check for null.